core/psh/uptime/uptime.c

```diff
@@ -27,12 +27,37 @@ void psh_uptimeinfo(void)
 int psh_uptime(int argc, char **argv)
 {
 	struct timespec tp;
-	int rv = clock_gettime(CLOCK_MONOTONIC_RAW, &tp);
+	intmax_t days, seconds;
+	int minutes, hours, rv;
```

Comment:
```suggestion
	unsigned int days, seconds;
	unsigned int minutes, hours;
	int rv;
```
Maybe that's enough

Reply:
`int`s are okay here, the `printf`s have `%d`, not `%u`. Variables: `seconds` should be `intmax_t` as `tp.tv_sec` is `intmax_t` (see POSIX), then `days` must be `intmax_t` too because it may be grater than `int`. I agree that `minutes` and `hours` could be `unsigned`, even `unsigned char` may be good because `minutes` are always in `<0, 59>` range and `hours` always in `<0,23>` range. But let's stick to `int`s and `intmax_t`s for now.

Reply:
Days > 2^32
I wish you such an uptime.